optaplanner-core/src/main/java/org/optaplanner/core/impl/statistic/BestScoreStatistic.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.statistic;
+
+import java.util.function.Consumer;
+
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.config.solver.metric.SolverMetric;
+import org.optaplanner.core.impl.score.definition.ScoreDefinition;
+import org.optaplanner.core.impl.solver.DefaultSolver;
+
+public class BestScoreStatistic implements Consumer<Solver> {
```

Comment:
Instead of implementing `Consumer<Solver>`, which is a generic interface, I'd create our own interface.
This is a very specific functionality and we want to see where it is used in the code. `Consumer<Solver>` doesn't let us do this, but `SolverStatistic<Solver>` would.